server/src/main/java/org/elasticsearch/action/search/FetchSearchPhase.java

```diff
@@ -72,31 +78,13 @@ final class FetchSearchPhase extends SearchPhase {
         this.nextPhaseFactory = nextPhaseFactory;
         this.context = context;
         this.logger = context.getLogger();
-        this.resultConsumer = resultConsumer;
         this.progressListener = context.getTask().getProgressListener();
+        this.reducedQueryPhase = reducedQueryPhase;
     }
 
     @Override
     public void run() {
-        context.execute(new AbstractRunnable() {
-            @Override
-            protected void doRun() throws Exception {
-                // we do the heavy lifting in this inner run method where we reduce aggs etc. that's why we fork this phase
-                // off immediately instead of forking when we send back the response to the user since there we only need
-                // to merge together the fetched results which is a linear operation.
```

Comment:
While this comment implies that this is the ONLY reason for forking to the context executable, this comment was made 7 years ago. I would be more comfortable if you added back the fork to the `execute` here. 

I suspect some other things have snuck into the fetch phase that we wouldn't want executing on the same thread as the caller.